src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php

```diff
@@ -116,22 +112,36 @@ public function extractContext($trace, $class)
     }
 
     /**
-     * Attach suppressed exception thrown before _after hook to the current step.
+     * Attach stack trace of exceptions thrown in each test hook to allure.
      * @param  \Exception $exception
+     * @param  string     $testMethod
      * @return mixed
      */
-    public function logPreviousException(\Exception $exception)
+    public function attachExceptionToAllure($exception, $testMethod)
     {
+        if (is_subclass_of($exception, \PHPUnit\Framework\Exception::class)) {
+            $trace = $exception->getSerializableTrace();
+        } else {
+            $trace = $exception->getTrace();
+        }
+
+        $context = $this->extractContext($trace, $testMethod);
```

Comment:
How do you feel about doing a mapping here?
```
const TEST_HOOKS = ['_after' => 'AfterHook']
...
if (isset(self::TEST_HOOKS[$context])) {
    $context = self::TEST_HOOKS[$context];
} else {
    $context = 'Test'
}
```

Reply:
Added mappings, modified 'Test' to 'TestMethod' as per Cest.php verbiage. Open to changing it back to 'Test' if you think it'll be better.
![image](https://user-images.githubusercontent.com/8500612/69173745-01e7eb00-0ac6-11ea-99ea-30dd7192c385.png)